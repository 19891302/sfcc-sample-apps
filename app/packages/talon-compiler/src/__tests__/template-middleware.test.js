const { generateHTML, templateMiddleware } = require('../template-middleware');
const { startTestContext, endTestContext } = require('test-talon-context');
const resourceService = require('../resources/resource-service');
const { getOutputConfigs } = require('talon-common');

jest.mock('../resources/resource-service');
jest.mock('../utils/observable-folder-hash');

const versionKey = '1';

beforeEach(() => {
    return startTestContext({ versionKey });
});

afterEach(() => {
    endTestContext();
});

describe('template-middleware', () => {
    describe('generateHTML', () => {
        getOutputConfigs().forEach(({ mode }) => {
            it(`generates HTML in ${mode} mode`, async () => {
                return generateHTML(mode, '/my-page').then(html => {
                    expect(html).toMatchSnapshot();
                });
            });
        });

        it('includes UIDs', async () => {
            const mode = 'dev';
            resourceService.get.mockImplementation(descriptor => {
                const uids = {
                    dev: `[${mode} uid for ${descriptor}]`
                };
                return Promise.resolve({ descriptor, uids });
            });

            return generateHTML(mode, '/').then(html => {
                expect(html).toMatchSnapshot();
            });
        });
    });

    describe('templateMiddleware', () => {
        it('send HTML', () => {
            const middleware = templateMiddleware();

            const req = {
                query: {
                    mode: 'dev'
                },
                originalUrl: '/my-page?a=1'
            };

            const res = {
                send: jest.fn()
            };

            function next(err) {
                throw err;
            }

            return middleware(req, res, next).then(() => {
                expect(res.send).toHaveBeenCalledTimes(1);
                expect(res.send.mock.calls[0][0]).toMatchSnapshot();
            });
        });
    });
});