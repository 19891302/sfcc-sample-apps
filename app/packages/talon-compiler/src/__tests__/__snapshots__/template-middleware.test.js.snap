// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template-middleware generateHTML generates HTML in dev mode 1`] = `
"<script src=\\"/basePath/talon/framework/latest/dev/talon.js\\"></script>
<script type=\\"text/javascript\\">
    Talon.configProvider.register({
        getBasePath() {
            return \\"/basePath\\";
        },
        getMode() {
            return \\"dev\\";
        },
        getLocale() {
            return \\"en_US\\";
        }
    });

    Talon.themeService.setTheme({\\"themeLayouts\\":{\\"inner\\":{\\"view\\":\\"themeLayoutInner\\"},\\"default\\":{\\"view\\":\\"themeLayoutDefault\\"}}});
Talon.themeService.setViewToThemeLayoutMap({\\"home\\":\\"themeLayoutInner\\",\\"myPage\\":\\"themeLayoutInner\\",\\"error\\":\\"themeLayoutDefault\\"});
Talon.routingService.registerRoutes([{\\"name\\":\\"home\\",\\"path\\":\\"/\\",\\"isRoot\\":true,\\"view\\":\\"home\\"},{\\"name\\":\\"my-page\\",\\"path\\":\\"/my-page\\",\\"view\\":\\"myPage\\"},{\\"name\\":\\"error\\",\\"path\\":\\"/error\\",\\"isDefault\\":true,\\"view\\":\\"error\\"}]);
Talon.brandingService.setBrandingProperties({});
</script>
<script src=\\"/basePath/talon/view/latest/dev/en_US/themeLayoutInner.js\\"></script>
<script src=\\"/basePath/talon/view/latest/dev/en_US/myPage.js\\"></script>


<script type=\\"text/javascript\\">
    const container = document.getElementById(\\"main\\");
    Talon.componentService.createElement(\\"talon/app\\")
        .then(element => {
            container.appendChild(element);
        });
    </script>

Base path: /basePath
Version key: 1"
`;

exports[`template-middleware generateHTML generates HTML in prod mode 1`] = `
"<script src=\\"/basePath/talon/framework/latest/prod/talon.js\\"></script>
<script type=\\"text/javascript\\">
    Talon.configProvider.register({
        getBasePath() {
            return \\"/basePath\\";
        },
        getMode() {
            return \\"prod\\";
        },
        getLocale() {
            return \\"en_US\\";
        }
    });

    Talon.themeService.setTheme({\\"themeLayouts\\":{\\"inner\\":{\\"view\\":\\"themeLayoutInner\\"},\\"default\\":{\\"view\\":\\"themeLayoutDefault\\"}}});
Talon.themeService.setViewToThemeLayoutMap({\\"home\\":\\"themeLayoutInner\\",\\"myPage\\":\\"themeLayoutInner\\",\\"error\\":\\"themeLayoutDefault\\"});
Talon.routingService.registerRoutes([{\\"name\\":\\"home\\",\\"path\\":\\"/\\",\\"isRoot\\":true,\\"view\\":\\"home\\"},{\\"name\\":\\"my-page\\",\\"path\\":\\"/my-page\\",\\"view\\":\\"myPage\\"},{\\"name\\":\\"error\\",\\"path\\":\\"/error\\",\\"isDefault\\":true,\\"view\\":\\"error\\"}]);
Talon.brandingService.setBrandingProperties({});
</script>
<script src=\\"/basePath/talon/view/latest/prod/en_US/themeLayoutInner.js\\"></script>
<script src=\\"/basePath/talon/view/latest/prod/en_US/myPage.js\\"></script>


<script type=\\"text/javascript\\">
    const container = document.getElementById(\\"main\\");
    Talon.componentService.createElement(\\"talon/app\\")
        .then(element => {
            container.appendChild(element);
        });
    </script>

Base path: /basePath
Version key: 1"
`;

exports[`template-middleware generateHTML includes UIDs 1`] = `
"<script src=\\"/basePath/talon/framework/[dev uid for framework://talon]/dev/talon.js\\"></script>
<script type=\\"text/javascript\\">
    Talon.configProvider.register({
        getBasePath() {
            return \\"/basePath\\";
        },
        getMode() {
            return \\"dev\\";
        },
        getLocale() {
            return \\"en_US\\";
        }
    });

    Talon.themeService.setTheme({\\"themeLayouts\\":{\\"inner\\":{\\"view\\":\\"themeLayoutInner\\"},\\"default\\":{\\"view\\":\\"themeLayoutDefault\\"}}});
Talon.themeService.setViewToThemeLayoutMap({\\"home\\":\\"themeLayoutInner\\",\\"myPage\\":\\"themeLayoutInner\\",\\"error\\":\\"themeLayoutDefault\\"});
Talon.routingService.registerRoutes([{\\"name\\":\\"home\\",\\"path\\":\\"/\\",\\"isRoot\\":true,\\"view\\":\\"home\\"},{\\"name\\":\\"my-page\\",\\"path\\":\\"/my-page\\",\\"view\\":\\"myPage\\"},{\\"name\\":\\"error\\",\\"path\\":\\"/error\\",\\"isDefault\\":true,\\"view\\":\\"error\\"}]);
Talon.brandingService.setBrandingProperties({});
Talon.moduleRegistry.setResourceUids({\\"framework://talon\\":\\"[dev uid for framework://talon]\\",\\"view://home@en_US\\":\\"[dev uid for view://home@en_US]\\",\\"view://themeLayoutInner@en_US\\":\\"[dev uid for view://themeLayoutInner@en_US]\\"});
</script>
<script src=\\"/basePath/talon/view/[dev uid for view://themeLayoutInner@en_US]/dev/en_US/themeLayoutInner.js\\"></script>
<script src=\\"/basePath/talon/view/[dev uid for view://home@en_US]/dev/en_US/home.js\\"></script>


<script type=\\"text/javascript\\">
    const container = document.getElementById(\\"main\\");
    Talon.componentService.createElement(\\"talon/app\\")
        .then(element => {
            container.appendChild(element);
        });
    </script>

Base path: /basePath
Version key: 1"
`;

exports[`template-middleware templateMiddleware send HTML 1`] = `
"<script src=\\"/basePath/talon/framework/[dev uid for framework://talon]/dev/talon.js\\"></script>
<script type=\\"text/javascript\\">
    Talon.configProvider.register({
        getBasePath() {
            return \\"/basePath\\";
        },
        getMode() {
            return \\"dev\\";
        },
        getLocale() {
            return \\"en_US\\";
        }
    });

    Talon.themeService.setTheme({\\"themeLayouts\\":{\\"inner\\":{\\"view\\":\\"themeLayoutInner\\"},\\"default\\":{\\"view\\":\\"themeLayoutDefault\\"}}});
Talon.themeService.setViewToThemeLayoutMap({\\"home\\":\\"themeLayoutInner\\",\\"myPage\\":\\"themeLayoutInner\\",\\"error\\":\\"themeLayoutDefault\\"});
Talon.routingService.registerRoutes([{\\"name\\":\\"home\\",\\"path\\":\\"/\\",\\"isRoot\\":true,\\"view\\":\\"home\\"},{\\"name\\":\\"my-page\\",\\"path\\":\\"/my-page\\",\\"view\\":\\"myPage\\"},{\\"name\\":\\"error\\",\\"path\\":\\"/error\\",\\"isDefault\\":true,\\"view\\":\\"error\\"}]);
Talon.brandingService.setBrandingProperties({});
Talon.moduleRegistry.setResourceUids({\\"framework://talon\\":\\"[dev uid for framework://talon]\\",\\"view://themeLayoutInner@en_US\\":\\"[dev uid for view://themeLayoutInner@en_US]\\",\\"view://myPage@en_US\\":\\"[dev uid for view://myPage@en_US]\\"});
</script>
<script src=\\"/basePath/talon/view/[dev uid for view://themeLayoutInner@en_US]/dev/en_US/themeLayoutInner.js\\"></script>
<script src=\\"/basePath/talon/view/[dev uid for view://myPage@en_US]/dev/en_US/myPage.js\\"></script>


<script type=\\"text/javascript\\">
    const container = document.getElementById(\\"main\\");
    Talon.componentService.createElement(\\"talon/app\\")
        .then(element => {
            container.appendChild(element);
        });
    </script>

Base path: /basePath
Version key: 1"
`;
